.github/workflows/ci.yaml

```diff
@@ -54,3 +54,8 @@ jobs:
     uses: ./.github/workflows/publish-test-pypi.yaml
     secrets:
       test-pypi-token: ${{ secrets.TEST_PYPI_TOKEN }}
+    
+  gen-docs:
+    name: Generate docs
+    needs: [test]
+    uses: ./.github/workflows/gen-docs.yaml
```

Comment:
You generate the docs after you publish to test pypi. It is not critical but I would say it is better to do it the other way around